radixdlt-java/src/main/java/com/radixdlt/client/application/translate/TokenFeeProcessor.java

```diff
@@ -59,22 +60,19 @@ public TokenFeeProcessor(RRI tokenRri, FeeTable feeTable) {
 	}
 
 	@Override
-	public void process(ActionProcessor actionProcessor, MetadataProcessor metadataProcessor, RadixAddress address, Atom atom) {
-		UInt256 feeToPay = feeFor(atom);
-		BurnTokensAction feeAction = BurnTokensAction.create(
-			this.tokenRri,
-			address,
-			TokenUnitConversions.subunitsToUnits(feeToPay)
-		);
-		actionProcessor.process(feeAction);
+	public void process(ActionProcessor actionProcessor, RadixAddress address, Atom atom, Optional<BigDecimal> optionalFee) {
+		BigDecimal feeToPay = optionalFee.orElseGet(() -> feeFor(atom));
+		if (feeToPay.signum() != 0) {
```

Comment:
I assume fee can't be negative. Do we need to add a check for that?

Reply:
Oh, yes, good point.